File: frontend/src/types/interpreter.ts
export type InterpreterStatus =
  | "Active"
  | "Inactive"
  | "On Hold"
  | "Fully Onboarded"
  | "Terminated"
  | "Deactived"
  | "Resigned";
export type PaymentFrequency = "Weekly" | "Biweekly" | "Monthly";

export interface InterpreterItem {
  id: number;
  employee_id: string;
  full_name: string;
  email: string;
  language: string;
  location: string;
  country: string;
  associated_client_id: number;
  associated_client_name: string;
  payment_frequency: PaymentFrequency;
  weekly: string;
  rate: string;
  status: InterpreterStatus;
  propio_interpreter_id: string;
  big_interpreter_id: string;
  equiti_voyce_id: string;
  equiti_martti_id: string;
  mercury_recipient_id: string;
  zoho_contact_id: string | null;
  last_synced_at: string | null;
  sync_status: string;
  sync_error_message: string | null;
  created_at: string;
}

export interface InterpreterClientOption {
  id: number;
  name: string;
}

export interface InterpreterMetaResponse {
  clients: InterpreterClientOption[];
  payment_frequency_options: PaymentFrequency[];
  status_options: InterpreterStatus[];
}

export interface InterpreterPayload {
  employee_id: string;
  full_name: string;
  email: string;
  language: string;
  location: string;
  country: string;
  client_id: number;
  payment_frequency: PaymentFrequency;
  weekly: string;
  rate: string;
  status: InterpreterStatus;
  propio_interpreter_id: string;
  big_interpreter_id: string;
  equiti_voyce_id: string;
  equiti_martti_id: string;
  mercury_recipient_id: string;
}

export interface InterpreterFilters {
  search: string;
  full_name: string;
  employee_id: string;
  language: string;
  location: string;
  country: string;
  client_id: string;
  payment_frequency: string;
  status: string;
}

export interface InterpreterSyncResponse {
  total_fetched: number;
  eligible: number;
  created: number;
  updated: number;
  skipped: number;
  errors: number;
  message: string;
}

export interface InterpreterSyncStatusResponse {
  last_full_sync_at: string | null;
  last_sync_status: string;
  last_sync_error_message: string | null;
  synced_records: number;
  error_records: number;
}

export interface MercuryRecipientSyncResponse {
  success: true;
  totalRecipients: number;
  matched: number;
  updated: number;
  skippedExisting: number;
  unmatched: number;
  duplicates: number;
  unmatchedEmails: string[];
  duplicateEmails: string[];
  errors: string[];
}

export interface InterpreterListResponse {
  items: InterpreterItem[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: backend/app/schemas/interpreter.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.models.enums import InterpreterStatus, PaymentFrequency


class InterpreterBase(BaseModel):
    employee_id: str = Field(min_length=1, max_length=64)
    full_name: str = Field(min_length=1, max_length=255)
    email: EmailStr
    language: str = Field(min_length=1, max_length=120)
    location: str = Field(min_length=1, max_length=120)
    country: str = Field(min_length=1, max_length=120)
    client_id: int = Field(gt=0)
    payment_frequency: PaymentFrequency
    rate: Decimal = Field(ge=0)
    status: InterpreterStatus
    propio_interpreter_id: str = ""
    big_interpreter_id: str = ""
    equiti_voyce_id: str = ""
    equiti_martti_id: str = ""
    mercury_recipient_id: str = ""


class InterpreterCreate(InterpreterBase):
    pass


class InterpreterUpdate(InterpreterBase):
    pass


class InterpreterListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    employee_id: str
    full_name: str
    email: EmailStr
    language: str
    location: str
    country: str
    associated_client_id: int
    associated_client_name: str
    payment_frequency: PaymentFrequency
    rate: Decimal
    status: InterpreterStatus
    propio_interpreter_id: str = ""
    big_interpreter_id: str = ""
    equiti_voyce_id: str = ""
    equiti_martti_id: str = ""
    mercury_recipient_id: str = ""
    zoho_contact_id: str | None = None
    last_synced_at: datetime | None = None
    sync_status: str
    sync_error_message: str | None = None
    created_at: datetime


class InterpreterClientOption(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str


class InterpreterMetaResponse(BaseModel):
    clients: list[InterpreterClientOption]
    payment_frequency_options: list[PaymentFrequency]
    status_options: list[InterpreterStatus]
